pennylane/plugins/default_qubit.py

```diff
@@ -521,11 +521,16 @@ def operations(self):
     def observables(self):
         return set(self._observable_map.keys())
 
-    def probability(self):
+    def probability(self, wires=None):
         if self._state is None:
             return None
 
-        states = itertools.product(range(2), repeat=self.num_wires)
-        probs = np.abs(self._state)**2
+        prob = np.abs(self._state.reshape([2] * self.num_wires)) ** 2
 
-        return OrderedDict(zip(states, probs))
+        wires = wires or range(self.num_wires)
+        wires = np.hstack(wires)
```

Comment:
How come this line would be needed? One of the reasons could be flattening ``wires`` here, however, would it not be required to have ``wires`` already as a flat ``list`` or an ``array``? (Also this performs the implicit conversion on ``range(self.num_wires)``)

Reply:
Good question. I can't remember why I needed to add this line :thinking: